routing/pathfind.go

```diff
@@ -298,34 +293,25 @@ func newRoute(amtToSend, feeLimit lnwire.MilliSatoshi, sourceVertex Vertex,
 		// If this is the last hop, then we send the exact amount and
 		// pay no fee, as we're paying directly to the receiver, and
 		// there're no additional hops.
-		amtToForward := runningAmt
+		amtToForward := amtToSend
 		fee := lnwire.MilliSatoshi(0)
 
 		// If this isn't the last hop, to add enough funds to pay for
 		// transit over the next link.
 		if i != len(pathEdges)-1 {
 			// We'll grab the edge policy and per-hop payload of
 			// the prior hop so we can calculate fees properly.
-			prevEdge := pathEdges[i+1]
-			prevHop := route.Hops[i+1]
-
-			// The fee for this hop, will be based on how much the
-			// prior hop carried, as we'll need to increase the
-			// amount of satoshis incoming into this hop to
-			// properly pay the required fees.
-			prevAmount := prevHop.AmtToForward
-			fee = computeFee(prevAmount, prevEdge.ChannelEdgePolicy)
-
-			// With the fee computed, we increment the total amount
-			// as we need to pay this fee. This value represents
-			// the amount of funds that will come _into_ this edge.
-			runningAmt += fee
-
-			// Otherwise, for a node to forward an HTLC, then
-			// following inequality most hold true:
-			//
-			//     * amt_in - fee >= amt_to_forward
-			amtToForward = runningAmt - fee
+			nextHop := route.Hops[i+1]
+
+			// The amount that this hop needs to forward is based
+			// on how much the prior hop carried plus the fee
+			// that needs to be paid to the prior hop
+			amtToForward = nextHop.AmtToForward + nextHop.Fee
```

Comment:
The fee has already been accumulated into what the next hop pays. The fee is the difference between hop_{i+1}.amount and hop_{i}.amount.

Reply:
I think what is confusing here is how the `Hop.Fee` relates to the `Hop.AmountToForward`. From the godoc on those fields, it looks like the end node of this `Hop` (=edge) will receive `AmountToFoward`, meaning the start node subtracted `Fee`. But here we set the "last hop" fee to zero, which doesn't work with that definition. So I think we should make the documentation match the usage here (by either modifying docs or use).

Reply:
A hop is a node according to lightning-rfc. I think we should stick to that definition in the code?

Reply:
I think it makes sense to name `hop=node` with the current comments :)